### PR TITLE
Created CONTRIBUTOR.md file

### PR DESCRIPTION
**Pull request by tannerjones4075**
## Description

Create a CONTRIBUTOR.md file to help developers understand, contribute, and test.

### CODE REVIEW

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:46]: I think this project is Apache 2.0. Can you please double-check?

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:3]: `bom-squad` is the GitHub oganization. `protobom` is the specific project.

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:7]: Same thing as above. `protobom` is the repository.

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:20]: Same as above. `protobom` is the repository.

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:25]: Are there development docs yet? If not, perhaps just add the word "forthcoming."

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:31]: I defer to true Go programmers here. @puerco, @manifestori, @houdini91. Any thoughts here?

File: CONTRIBUTOR.md
Comment (jspeed-meyers) [inline CONTRIBUTOR.md:42]: I would use the issues link for the `protobom` project, not for the overall GitHub organization.

**Review by jspeed-meyers — CHANGES_REQUESTED** [review]

**Comment by jspeed-meyers** [comment]
Thank you, @tannerjones4075. Some minor comments!

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:46]: Updated

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:3]: Done

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:7]: done

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:20]: done

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:25]: There is a developer.md file, though I am sure it could use some updating. 
```
# Development

### Development Container

1. Docker / Podman
2. VSCode (or other supported IDE)
3. Remote-Containers [extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)

### Native

1. Go 1.20+
2. Make

## Getting Started

First, clone this repository and make sure you have your enviornment all setup. </br>
For support IDEs users, you can simply run this project in a container (aka devcontainer): </br>

The option would be available once you install the recommended extension. </br>

See more details [here](https://containers.dev/supporting)
```

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:42]: done

**Review by jspeed-meyers — APPROVED** [review]
LGTM. Thank you, @tannerjones4075!

File: CONTRIBUTOR.md
Comment (puerco) [inline CONTRIBUTOR.md:38]: Conformance tests are separate from unit/integration tests. Perhaps we should go into what each does (happy to help adding some blurb here)

File: CONTRIBUTOR.md
Comment (puerco) [reply CONTRIBUTOR.md:31]: Sounds good to me 👍

File: CONTRIBUTOR.md
Comment (puerco) [inline CONTRIBUTOR.md:46]:
```suggestion
By contributing to the protobom project, you agree that your contributions will be licensed under the [Apache 2.0 License](LICENSE).
```

Comment (puerco) [reply CONTRIBUTOR.md:46]: This should also mention that commits need to have a signoff, at some point we will run a DCO check.

**Review by puerco — COMMENTED** [review]
Thanks for creating the contribution doc @tannerjones4075  this is a good starting point to get things going, I added a few notes, but (at least in my view) we could merge it and start iterating on it.

File: CONTRIBUTOR.md
Comment (tannerjones4075) [reply CONTRIBUTOR.md:38]: Should there be test/unit_testing/README.md added to the repo? All I see under testing is conformance.

Comment (puerco) [reply CONTRIBUTOR.md]: Not really, I think the regular tests should be what regular go contributors expect. The conformance tests are about ensuring the integrity of the data, which is why we keep a separate README. Just mentioning in the contributing.md file that we strive to keep our test coverage high so please add tests to any functions you introduce should be enough.  That and a link to the conformance testing.

**Comment by jspeed-meyers** [comment]
@puerco: Are you good with merging this?

**Review by puerco — APPROVED** [review]
OK, lgtm :rocket:

**Pull request merged**
